lars-reimann/api-editor
# fix: Constant & Optional annotations should only parse specific value types

Closes #531.

### Summary of Changes

Fixed a bug, where non-string, non-number, non-bool and non-none values are not handled like a string for constant and optional annotations.

### Testing Instructions

1. Run this:
  ```sh
  cd package-parser
  poetry run parse-package all -p package_parser -s package_parser -c package_parser -o out
  ```
2. Check the created file `out/annotations.json`.
Output like the following should no longer be contained in the json.
  ```json5
  {
    "constants": {
      "package_parser/package_parser.model.api._api/Module/__init__/name": {
        "target": "package_parser/package_parser.model.api._api/Module/__init__/name",
        "defaultType": "string",
        "defaultValue": "module_node.qname()" // <------------------------
      },
    }
  }
  ```
